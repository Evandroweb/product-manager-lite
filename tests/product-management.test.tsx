'use client'

import { render } from '@testing-library/react'
import ProductManagement from '@/components/product-management'
import { describe, expect, it, jest } from '@jest/globals'

jest.mock('@/hooks/use-media-query', () => ({
  useMediaQuery: () => false
}))

jest.mock('@/store/product-store', () => ({
  useProductStore: () => ({
    products: [
      {
        id: '1',
        name: 'Smartphone Galaxy S23',
        category: 'EletrÃ´nicos',
        price: 4999.99,
        description: 'Smartphone Samsung Galaxy S23 com 256GB de armazenamento e 8GB de RAM.',
        imageUrl: 'https://placehold.co/400x400/png'
      },
      {
        id: '2',
        name: 'Notebook Dell Inspiron',
        category: 'Computadores',
        price: 3599.9,
        description: 'Notebook Dell Inspiron com processador Intel Core i5, 8GB de RAM e SSD de 256GB.',
        imageUrl: 'https://placehold.co/400x400/png'
      }
    ],
    editingProduct: null,
    addProduct: jest.fn(),
    updateProduct: jest.fn(),
    removeProduct: jest.fn(),
    setEditingProduct: jest.fn()
  })
}))

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => {
    const imgProps = { ...props }
    if (imgProps.fill === true) {
      imgProps.fill = 'true'
    }
    return <img {...imgProps} alt={props.alt || ''} />
  }
}))

jest.mock('@/components/ui/slider', () => ({
  Slider: ({ defaultValue, value, onValueChange }: any) => (
    <div data-testid='mock-slider' data-value={JSON.stringify(value || defaultValue)}>
      <button onClick={() => onValueChange && onValueChange([0, 500])}>Change Value</button>
    </div>
  )
}))

describe('ProductManagement', () => {
  it('should render correctly', () => {
    const { container } = render(<ProductManagement />)
    expect(container).toMatchSnapshot()
  })
})
