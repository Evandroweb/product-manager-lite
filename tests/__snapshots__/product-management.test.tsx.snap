// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductManagement should render correctly 1`] = `
<div>
  <div
    class="space-y-6"
  >
    <div
      class="grid grid-cols-1 lg:grid-cols-3 gap-6"
    >
      <div
        class="lg:col-span-2 space-y-4"
      >
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-sm"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <div
              class="text-2xl font-semibold leading-none tracking-tight"
            >
              Filtros
            </div>
          </div>
          <div
            class="p-6 pt-0 space-y-4"
          >
            <div
              class="space-y-2"
            >
              <label
                class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                for="name-filter"
              >
                Nome
              </label>
              <input
                class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
                id="name-filter"
                placeholder="Filtrar por nome"
                value=""
              />
            </div>
            <div
              class="space-y-2"
            >
              <div
                class="flex justify-between"
              >
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                >
                  Faixa de Preço
                </label>
                <span
                  class="text-sm text-muted-foreground"
                >
                  R$ 0,00
                   - 
                  R$ 5.000,00
                </span>
              </div>
              <div
                data-testid="mock-slider"
                data-value="[0,5000]"
              >
                <button>
                  Change Value
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="space-y-4"
        >
          <div
            class="rounded-md border"
          >
            <div
              class="relative w-full overflow-auto"
            >
              <table
                class="w-full caption-bottom text-sm"
              >
                <thead
                  class="[&_tr]:border-b"
                >
                  <tr
                    class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                  >
                    <th
                      class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 w-[100px]"
                    >
                      Imagem
                    </th>
                    <th
                      class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                    >
                      <button
                        class="flex items-center"
                      >
                        Nome
                        <svg
                          class="lucide lucide-arrow-up ml-2 h-4 w-4"
                          fill="none"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="m5 12 7-7 7 7"
                          />
                          <path
                            d="M12 19V5"
                          />
                        </svg>
                      </button>
                    </th>
                    <th
                      class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                    >
                      <button
                        class="flex items-center"
                      >
                        Categoria
                        <svg
                          class="lucide lucide-arrow-up-down ml-2 h-4 w-4"
                          fill="none"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="m21 16-4 4-4-4"
                          />
                          <path
                            d="M17 20V4"
                          />
                          <path
                            d="m3 8 4-4 4 4"
                          />
                          <path
                            d="M7 4v16"
                          />
                        </svg>
                      </button>
                    </th>
                    <th
                      class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                    >
                      <button
                        class="flex items-center"
                      >
                        Preço
                        <svg
                          class="lucide lucide-arrow-up-down ml-2 h-4 w-4"
                          fill="none"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="m21 16-4 4-4-4"
                          />
                          <path
                            d="M17 20V4"
                          />
                          <path
                            d="m3 8 4-4 4 4"
                          />
                          <path
                            d="M7 4v16"
                          />
                        </svg>
                      </button>
                    </th>
                    <th
                      class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 w-[150px]"
                    >
                      Ações
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="[&_tr:last-child]:border-0"
                >
                  <tr
                    class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                  >
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                    >
                      <div
                        class="relative h-12 w-12 overflow-hidden rounded-md"
                      >
                        <img
                          alt="Notebook Dell Inspiron"
                          class="object-cover"
                          fill="true"
                          src="https://placehold.co/400x400/png"
                        />
                      </div>
                    </td>
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0 font-medium"
                    >
                      Notebook Dell Inspiron
                    </td>
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                    >
                      Computadores
                    </td>
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                    >
                      R$ 3.599,90
                    </td>
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                    >
                      <div
                        class="flex space-x-2"
                      >
                        <button
                          class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
                        >
                          <svg
                            class="lucide lucide-pencil h-4 w-4 mr-2"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M17 3a2.85 2.83 0 1 1 4 4L7.5 20.5 2 22l1.5-5.5Z"
                            />
                            <path
                              d="m15 5 4 4"
                            />
                          </svg>
                          Editar
                        </button>
                        <button
                          class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
                        >
                          <svg
                            class="lucide lucide-trash2 h-4 w-4 mr-2"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M3 6h18"
                            />
                            <path
                              d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                            />
                            <path
                              d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                            />
                            <line
                              x1="10"
                              x2="10"
                              y1="11"
                              y2="17"
                            />
                            <line
                              x1="14"
                              x2="14"
                              y1="11"
                              y2="17"
                            />
                          </svg>
                          Excluir
                        </button>
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                  >
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                    >
                      <div
                        class="relative h-12 w-12 overflow-hidden rounded-md"
                      >
                        <img
                          alt="Smartphone Galaxy S23"
                          class="object-cover"
                          fill="true"
                          src="https://placehold.co/400x400/png"
                        />
                      </div>
                    </td>
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0 font-medium"
                    >
                      Smartphone Galaxy S23
                    </td>
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                    >
                      Eletrônicos
                    </td>
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                    >
                      R$ 4.999,99
                    </td>
                    <td
                      class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                    >
                      <div
                        class="flex space-x-2"
                      >
                        <button
                          class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
                        >
                          <svg
                            class="lucide lucide-pencil h-4 w-4 mr-2"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M17 3a2.85 2.83 0 1 1 4 4L7.5 20.5 2 22l1.5-5.5Z"
                            />
                            <path
                              d="m15 5 4 4"
                            />
                          </svg>
                          Editar
                        </button>
                        <button
                          class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
                        >
                          <svg
                            class="lucide lucide-trash2 h-4 w-4 mr-2"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M3 6h18"
                            />
                            <path
                              d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                            />
                            <path
                              d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                            />
                            <line
                              x1="10"
                              x2="10"
                              y1="11"
                              y2="17"
                            />
                            <line
                              x1="14"
                              x2="14"
                              y1="11"
                              y2="17"
                            />
                          </svg>
                          Excluir
                        </button>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <div
            class="flex items-center justify-between"
          >
            <div
              class="flex items-center space-x-2"
            >
              <p
                class="text-sm text-muted-foreground"
              >
                Exibindo 
                2
                 de 
                2
                 produtos
              </p>
              <button
                aria-autocomplete="none"
                aria-controls="radix-:r0:"
                aria-expanded="false"
                class="flex items-center justify-between rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 h-8 w-[70px]"
                data-state="closed"
                dir="ltr"
                role="combobox"
                type="button"
              >
                <span
                  style="pointer-events: none;"
                >
                  5
                </span>
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </div>
            <div
              class="flex items-center space-x-2"
            >
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
                disabled=""
              >
                <svg
                  class="lucide lucide-chevron-left h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m15 18-6-6 6-6"
                  />
                </svg>
              </button>
              <span
                class="text-sm"
              >
                Página 
                1
                 de 
                1
              </span>
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
                disabled=""
              >
                <svg
                  class="lucide lucide-chevron-right h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m9 18 6-6-6-6"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="lg:col-span-1"
      >
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-sm"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <div
              class="text-2xl font-semibold leading-none tracking-tight"
            >
              Novo Produto
            </div>
          </div>
          <div
            class="p-6 pt-0"
          >
            <form
              class="space-y-4"
            >
              <div
                class="space-y-2"
              >
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                  for=":r1:-form-item"
                >
                  Nome
                </label>
                <input
                  aria-describedby=":r1:-form-item-description"
                  aria-invalid="false"
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
                  id=":r1:-form-item"
                  name="name"
                  placeholder="Nome do produto"
                  value=""
                />
              </div>
              <div
                class="space-y-2"
              >
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                  for=":r2:-form-item"
                >
                  Categoria
                </label>
                <input
                  aria-describedby=":r2:-form-item-description"
                  aria-invalid="false"
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
                  id=":r2:-form-item"
                  name="category"
                  placeholder="Categoria do produto"
                  value=""
                />
              </div>
              <div
                class="space-y-2"
              >
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                  for=":r3:-form-item"
                >
                  Preço
                </label>
                <input
                  aria-describedby=":r3:-form-item-description"
                  aria-invalid="false"
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
                  id=":r3:-form-item"
                  name="price"
                  placeholder="0,00"
                  value="0"
                />
              </div>
              <div
                class="space-y-2"
              >
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                  for=":r4:-form-item"
                >
                  Descrição
                </label>
                <textarea
                  aria-describedby=":r4:-form-item-description"
                  aria-invalid="false"
                  class="flex min-h-[80px] w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm resize-none"
                  id=":r4:-form-item"
                  name="description"
                  placeholder="Descrição do produto"
                />
              </div>
              <div
                class="space-y-2"
              >
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                  for=":r5:-form-item"
                >
                  URL da Imagem
                </label>
                <input
                  aria-describedby=":r5:-form-item-description"
                  aria-invalid="false"
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
                  id=":r5:-form-item"
                  name="imageUrl"
                  placeholder="https://exemplo.com/imagem.jpg"
                  value=""
                />
              </div>
              <div
                class="flex justify-end space-x-2"
              >
                <button
                  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
                  type="submit"
                >
                  Cadastrar Produto
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
